### PR TITLE
Fix center profile

### LINKED ISSUE #328: `CenterAndProfile` fails if the image has a background
It turns out that `profiles.CenterAndProfile` fails in many cases if the image has a background. It works if the star is bright enough (or anytime the peak is much, much higher than the background) but fails otherwise because it is fitting a Gaussian  to the profile that expects to go to zero at large enough radii.

PR incoming...

### LINKED ISSUE #330: `CenterAndProfile` does not return a background-subtracted radial profile
An incorrect profile is below. It turns out that none of our profile tests had a background....

lab_-_JupyterLab

### PR DESCRIPTION
**Pull request by mwcraig**
This fixes #328 by subtracting a rough background in the initial creation of the radial profile. It also fixes #330 by subtracting the background.

### CODE REVIEW

**Comment by mwcraig** [comment]
@JuanCab -- this is actually ready for a look now

File: stellarphot/photometry/profiles.py [hunk 1]
```diff
@@ -245,17 +247,32 @@ def pixel_values_in_profile(self):
...
         radii = []
         pixel_values = []
-        for rad, ap in zip(
-            self.radial_profile.radius, self.radial_profile.apertures, strict=True
-        ):
+        for ap in self.radial_profile.apertures:
+            # Calculate the distance of each pixel from the center
+            grid_x, grid_y = np.mgrid[: self._data.shape[0], : self._data.shape[1]]
+            dist_from_cen = np.sqrt(
+                (grid_x - self._cen[1]) ** 2 + (grid_y - self._cen[0]) ** 2
```
Comment (JuanCab) [inline stellarphot/photometry/profiles.py:254]: Is this correct, you have `grid_x` determined by `: self._data.shape[0]` but you compute the distance from center using `(grid_x - self._cen[1])`?

File: stellarphot/photometry/tests/test_profiles.py [hunk 2]
```diff
@@ -72,6 +72,8 @@ def test_find_center_no_star():
 
 
 def test_radial_profile():
+    # Test that both curve of growth and radial profile are correct
+
```
Comment (JuanCab) [inline stellarphot/photometry/tests/test_profiles.py:76]: Looks like a reasonable test

File: stellarphot/photometry/tests/test_profiles.py [hunk 3]
```diff
@@ -102,3 +112,32 @@ def test_radial_profile_exposure_is_nan():
     assert np.isnan(rad_prof.noise(c, np.nan)[-1])
     assert all(np.isfinite(rad_prof.curve_of_growth.profile))
     assert all(np.isfinite(rad_prof.radial_profile.profile))
+
+
+def test_radial_profile_with_background():
+    # Regression test for #328 -- image with a background level
+    image = make_gaussian_sources_image(SHAPE, STARS)
+    image = image + +make_noise_image(
```
Comment (JuanCab) [inline stellarphot/photometry/tests/test_profiles.py:120]: Superfluous + sign, I think...

File: stellarphot/photometry/tests/test_profiles.py [hunk 4]
```diff
@@ -102,3 +112,32 @@ def test_radial_profile_exposure_is_nan():
     assert np.isnan(rad_prof.noise(c, np.nan)[-1])
     assert all(np.isfinite(rad_prof.curve_of_growth.profile))
     assert all(np.isfinite(rad_prof.radial_profile.profile))
+
+
+def test_radial_profile_with_background():
+    # Regression test for #328 -- image with a background level
```
Comment (JuanCab) [inline stellarphot/photometry/tests/test_profiles.py:118]: Again, I believe the logic here is sound, though it took a bit for me to understand it.

**Review by JuanCab — DISMISSED** [review]
Looks good... but check my comment about the computation of `list_from_cen` and make sure that is correct...

File: stellarphot/photometry/profiles.py (hunk 1 above)
Comment (mwcraig) [reply stellarphot/photometry/profiles.py:254]: Yeah; this is the array vs image thing.

File: stellarphot/photometry/tests/test_profiles.py (hunk 3 above)
Comment (mwcraig) [reply stellarphot/photometry/tests/test_profiles.py]: Fixed

**Comment by mwcraig** [comment]
I fixed the "+" sign and double-checked the `1`/`0` index thing to confirm it is correct, going to merge.

**Pull request merged**